File: UI/src/app/domain/order/equipment.ts
import { EquipmentBase } from "../equipment-base";

export class Equipment extends EquipmentBase {
    availableComplectation: string[];
    isNode:boolean;
    parentId?:number;

    constructor(id:number, name:string, isNode:boolean, parentId:number, availableComplectation: string[]){
            super(id, name);
            this.availableComplectation=availableComplectation;
            this.isNode=isNode;
            this.parentId=parentId;
        }
}

File: Api/Reception.Api/Application/Reception.Domain/Entities/Equipment.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace Reception.Domain.Entities
{
    /// <summary>
    /// Оборудование.
    /// </summary>
    public class Equipment
    {
        public Equipment()
        {
            Orders = new HashSet<Order>();
            Complectations = new HashSet<Complectation>();
        }

        /// <summary>
        /// Идентификатор записи.
        /// </summary>
        public int Id { get; set; }

        /// <summary>
        /// Ссылка на родительскую запись.
        /// </summary>
        public int? ParentId { get; set; }

        /// <summary>
        /// Признак группы.
        /// </summary>
        public bool IsNode { get; set; }

        /// <summary>
        /// Наименование.
        /// </summary>
        public string Name { get; set; }

        public virtual ICollection<Order> Orders { get; set; }
        public virtual ICollection<Complectation> Complectations { get; set; }
    }
}
